ui/pages/SummaryData/reducers.js

```diff
@@ -54,13 +53,10 @@ export const loadSavedVariants = ({ tag, gene = '' }) => (dispatch, getState) =>
   dispatch({ type: REQUEST_SAVED_VARIANTS })
   new HttpRequestHelper(`/api/summary_data/saved_variants/${tag}`,
     (responseJson) => {
-      if (tag && !gene) {
-        dispatch({
-          type: RECEIVE_SAVED_VARIANT_TAGS,
-          updates: { [tag]: true },
-        })
-      }
-      dispatch({ type: RECEIVE_DATA, updatesById: responseJson })
+      dispatch({
+        type: RECEIVE_DATA,
+        updatesById: { ...responseJson, multiTagVariants: { [tag]: Object.keys(responseJson.savedVariantsByGuid) } },
```

Comment:
`if (tag && !gene) ` still seems like an important conditional, why was it removed?

Reply:
I proposed naming this `multiTagVariants` because I thought this would be special-cased only for multi variant tags, but this implementation is nicer. However, I would rename the key `savedVariantsByTag`

Reply:
> `if (tag && !gene)`  still seems like an important conditional, why was it removed?

Not do caching when there is a gene? It makes sense. I could add the condition to the value of the `savedVariantsByTag` generation. But it still has corner cases for switching from no gene to having a gene.

Reply:
How is that a corner case? The filtering in the selector will work fine for that

Reply:
I want to add the `gene` to the key to solve the issues. So the `savedVariantsByTag` will become `savedVariantsByTagGene`.

Reply:
the amount we are caching is overkill. This is too complex and it will save basically no time for the user. Do not add this additional level of caching. Just do this like search results, where theres a single state key with a list (for search its `searchedVariants`, here it will be something like `summaryDataVariants`) that just gets overwritten every time new summary variants are loaded, and then the selector checks that first and uses that if its there